test: fix 6 failing config/context tests on Windows (HOME vs USERPROFILE)

## Problem

Running `go test ./cmd/...` produces **6 failing tests** due to a Windows-specific env-var issue:

```
--- FAIL: TestLoadConfig
--- FAIL: TestSaveConfig
--- FAIL: TestConfigPath
--- FAIL: TestLoadContext
--- FAIL: TestSaveContext
--- FAIL: TestGetCurrentProjectID
```

### Root cause

`GetConfigPath()` and the context path helper both call `os.UserHomeDir()`. On Windows, `os.UserHomeDir()` reads `USERPROFILE` first (not `HOME`). The failing tests redirect the home directory by setting only `HOME`, so the functions still read the real user home and find (or not find) real files — causing false failures.

`cmd/root_test.go` already handles this correctly: it sets **both** `HOME` and `USERPROFILE`. The failing tests in `config_test.go` and `context_test.go` only set `HOME`.

### Fix

In every test in `config_test.go` and `context_test.go` that sets `HOME`, also set `USERPROFILE`:

```go
os.Setenv("HOME", tmpDir)
os.Setenv("USERPROFILE", tmpDir)           // add this line
defer os.Setenv("USERPROFILE", originalUserProfile)
```

This makes the test pattern consistent with `root_test.go`.

### Files to change

- `cmd/config_test.go` — 3 tests: `TestLoadConfig`, `TestSaveConfig`, `TestConfigPath`
- `cmd/context_test.go` — 3 tests: `TestLoadContext`, `TestSaveContext`, `TestGetCurrentProjectID`

### Acceptance criteria

- `go test ./cmd/...` passes with 0 failures on Windows, Linux, and macOS
- No real user home directory is read or written during the test run

--- a/cmd/config_test.go
+++ b/cmd/config_test.go
@@ -7,18 +7,17 @@ import (
 )
 
 func TestLoadConfig(t *testing.T) {
-	// Save original home dir
 	originalHome := os.Getenv("HOME")
-	if originalHome == "" {
-		originalHome = os.Getenv("USERPROFILE") // Windows
-	}
+	originalUserProfile := os.Getenv("USERPROFILE")
 
-	// Create temporary directory for test
 	tmpDir := t.TempDir()
 
-	// Set HOME to temp directory
 	os.Setenv("HOME", tmpDir)
-	defer os.Setenv("HOME", originalHome)
+	os.Setenv("USERPROFILE", tmpDir)
+	defer func() {
+		os.Setenv("HOME", originalHome)
+		os.Setenv("USERPROFILE", originalUserProfile)
+	}()
 
 	// Test with non-existent config
 	config, err := LoadConfig()
@@ -31,18 +30,17 @@ func TestLoadConfig(t *testing.T) {
 }
 
 func TestSaveConfig(t *testing.T) {
-	// Save original home dir
 	originalHome := os.Getenv("HOME")
-	if originalHome == "" {
-		originalHome = os.Getenv("USERPROFILE") // Windows
-	}
+	originalUserProfile := os.Getenv("USERPROFILE")
 
-	// Create temporary directory for test
 	tmpDir := t.TempDir()
 
-	// Set HOME to temp directory
 	os.Setenv("HOME", tmpDir)
-	defer os.Setenv("HOME", originalHome)
+	os.Setenv("USERPROFILE", tmpDir)
+	defer func() {
+		os.Setenv("HOME", originalHome)
+		os.Setenv("USERPROFILE", originalUserProfile)
+	}()
 
 	// Create test config
 	testConfig := &Config{
@@ -78,18 +76,17 @@ func TestSaveConfig(t *testing.T) {
 }
 
 func TestConfigPath(t *testing.T) {
-	// Save original home dir
 	originalHome := os.Getenv("HOME")
-	if originalHome == "" {
-		originalHome = os.Getenv("USERPROFILE") // Windows
-	}
+	originalUserProfile := os.Getenv("USERPROFILE")
 
-	// Create temporary directory for test
 	tmpDir := t.TempDir()
 
-	// Set HOME to temp directory
 	os.Setenv("HOME", tmpDir)
-	defer os.Setenv("HOME", originalHome)
+	os.Setenv("USERPROFILE", tmpDir)
+	defer func() {
+		os.Setenv("HOME", originalHome)
+		os.Setenv("USERPROFILE", originalUserProfile)
+	}()
 
 	// Get config path
 	configPath, err := GetConfigPath()

--- a/cmd/context_test.go
+++ b/cmd/context_test.go
@@ -7,18 +7,17 @@ import (
 )
 
 func TestLoadContext(t *testing.T) {
-	// Save original home dir
 	originalHome := os.Getenv("HOME")
-	if originalHome == "" {
-		originalHome = os.Getenv("USERPROFILE") // Windows
-	}
+	originalUserProfile := os.Getenv("USERPROFILE")
 
-	// Create temporary directory for test
 	tmpDir := t.TempDir()
 
-	// Set HOME to temp directory
 	os.Setenv("HOME", tmpDir)
-	defer os.Setenv("HOME", originalHome)
+	os.Setenv("USERPROFILE", tmpDir)
+	defer func() {
+		os.Setenv("HOME", originalHome)
+		os.Setenv("USERPROFILE", originalUserProfile)
+	}()
 
 	// Test with non-existent context
 	ctx, err := LoadContext()
@@ -31,18 +30,17 @@ func TestLoadContext(t *testing.T) {
 }
 
 func TestSaveContext(t *testing.T) {
-	// Save original home dir
 	originalHome := os.Getenv("HOME")
-	if originalHome == "" {
-		originalHome = os.Getenv("USERPROFILE") // Windows
-	}
+	originalUserProfile := os.Getenv("USERPROFILE")
 
-	// Create temporary directory for test
 	tmpDir := t.TempDir()
 
-	// Set HOME to temp directory
 	os.Setenv("HOME", tmpDir)
-	defer os.Setenv("HOME", originalHome)
+	os.Setenv("USERPROFILE", tmpDir)
+	defer func() {
+		os.Setenv("HOME", originalHome)
+		os.Setenv("USERPROFILE", originalUserProfile)
+	}()
 
 	// Create test context
 	testContext := &Context{
@@ -82,18 +80,17 @@ func TestSaveContext(t *testing.T) {
 }
 
 func TestGetCurrentProjectID(t *testing.T) {
-	// Save original home dir
 	originalHome := os.Getenv("HOME")
-	if originalHome == "" {
-		originalHome = os.Getenv("USERPROFILE") // Windows
-	}
+	originalUserProfile := os.Getenv("USERPROFILE")
 
-	// Create temporary directory for test
 	tmpDir := t.TempDir()
 
-	// Set HOME to temp directory
 	os.Setenv("HOME", tmpDir)
-	defer os.Setenv("HOME", originalHome)
+	os.Setenv("USERPROFILE", tmpDir)
+	defer func() {
+		os.Setenv("HOME", originalHome)
+		os.Setenv("USERPROFILE", originalUserProfile)
+	}()
 
 	// Test with no context
 	projectID, err := GetCurrentProjectID()

